### PR TITLE
activemodel: signature updates

### PR DESCRIPTION
**Pull request by hjwylde**
Split from #735 to be smaller. Type signature updates for invalid signatures in activemodel, one is due to the block being marked as required, but it should be optional. The other (`delete`) is updated to match the method declaration (see https://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-delete).

### CODE REVIEW

File: gems/activemodel/6.0/activemodel.rbs [hunk 1]
```diff
@@ -1,3 +1,14 @@
+module ActiveModel
+  class Errors
+    # Delete messages for +key+. Returns the deleted messages.
+    #
+    #   person.errors[:name]        # => ["cannot be nil"]
+    #   person.errors.delete(:name) # => ["cannot be nil"]
+    #   person.errors[:name]        # => []
+    def delete: (untyped key, ?untyped type, **untyped) -> untyped
```
[the remark this answers is not in the document]
Comment (hjwylde) [reply gems/activemodel/6.0/activemodel.rbs:8]: Hmm, out of curiosity, how does this work with 6.1 then? Will 6.1 use the types from 6.0, and is this acceptable for now, or should a directory for 6.1 be added?

[the remark this answers is not in the document]
Comment (tk0miya) [reply gems/activemodel/6.0/activemodel.rbs:8]: 6.x users will use the types for 6.0. So it's better to add a new type to the 6.1 directory. But Rails-6.1 is now outdated and not supported. Personally, it's not too late to add it after somebody wants it.

[the remark this answers is not in the document]
Comment (hjwylde) [reply gems/activemodel/6.0/activemodel.rbs]: Thanks for the check! I've fixed this up :).

[the remark this answers is not in the document]
Comment (tk0miya) [reply gems/activemodel/6.0/activemodel.rbs]: LGTM!

**Review by tk0miya — COMMENTED** [review]
APPROVE

**Pull request merged**